numpy/fft/_pocketfft.py

```diff
@@ -180,8 +180,10 @@ def fft(a, n=None, axis=-1, norm=None):
     if n is None:
         n = a.shape[axis]
     inv_norm = 1
-    if norm is not None and _unitary(norm):
+    if _unitary(norm):
```

Comment:
This is a bit confusing, the main thing the function seems to do is to validate the options, maybe we can just make the signature `_get_norm(norm, True/False)` (forward/backward) and then use `1./_get_norm(norm, True/False)` where necessary?
Or just `_check_norm` and only use this to raise an error.

Reply:
yes that's true; the reason I kept the `_unitary(norm)` function as is was to make sure I introduce minimal changes to the code. Up to now, there were only two `norm` options: `None` and `ortho`, thus a binary flag was enough to cover them both; now we move to three choices, so the function checks if `norm=ortho` or not (in agreement with its name); if not, then we need an additional check to distinguish between `norm=None` and `norm=forward`

Reply:
I totally agree that this might not be the most efficient way to do these checks, but seemed like the most straightforward change based on the existing codebase. I'd suggest that we keep the checks in their current state so that we pull the API change into NumPy as quickly as possible, so that CuPy and others can proceed as well. 

Then we can certainly check the internals of the code again and maybe streamline the checks or make them more straightforward; since the API will already be there, any downstream dependence will just update its code with no need for changes from their side, or coordination from our side. I'd be happy to keep working on the code if you think this sounds like a reasonable approach

Reply:
I would personally slightly prefer it here, since its a small change and the code is slightly odd right now, but OK (or is this private function used in CuPy somehow?).  The change should also get a `new_feature` news fragment, see: `doc/release/upcoming_changes/README.rst`

Reply:
@leofang @asi1024

Reply:
We don't use any private functions from `numpy.fft` in CuPy if it's the reason we are tagged 🙂

Reply:
yes that was exactly our question, thanks :)